### PR TITLE
feat: PBI 32 status page configuration frontend

### PR DESCRIPTION
**Pull request by HugoIr**
### Describe your changes
Implement frontend for status page configuration

### Backlog name
PBI-32: status page configuration

### Acceptance Criteria
- User can configure path of API status page
- User can create category for status page
- User can delete category for status page
- User can change status page category of each API monitor

### Preview Pages

Screen Shot 2022-12-04 at 10 48 28

When status page url path already exist
Screen Shot 2022-12-04 at 10 48 58

Add select category to create API monitor
Screen Shot 2022-12-08 at 14 02 25

Add select category to edit API monitor
Screen Shot 2022-12-08 at 14 02 45

Added category to API monitor is sync with preview page
Screen Shot 2022-12-08 at 14 15 05

### Checklist
- [x] I have performed a self-review of my code
- [x] Code successfully run on local
- [x] Feature / changes tested on local
- [x] No failing unit test
- [ ] Passed UAT Test
- [x] Sonarqube tested 
- [ ] Required any changes to environment variable

### CODE REVIEW

File: src/routes/statusPage/index.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+    useEffect(()=> {
+		axios.get(`${BASE_URL}/status-page/category/`, {
+			headers: {
+				Authorization:`Token ${getUserToken()}`
+			}
+		}).then((response)=>{
+			setCategoryList(response.data)
+		})
+    }, [categoryList])
```
Comment (andrew-susanto) [inline src/routes/statusPage/index.js]:
```suggestion
    }, [])
```

potential inifinite loop (?)

File: src/routes/statusPage/index.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                            <Box mb='30px' />
+
+                            <Text mb={23} fontSize='32px' fontWeight='semibold' color='black'>
+                            Status Page URL
+                            </Text>
+                            
+                            <Box mb='10px' />
+                            <Box as='form' onSubmit={handleSubmit(onSaveUrl)} id='form-save-url'>
+                                <Flex align='center'>
+                                    <Text>https://status.monapi.com/</Text>
```
Comment (andrew-susanto) [inline src/routes/statusPage/index.js]:
```suggestion
                                    <Text>{window.location.origin}/status/</Text>
```
kayaknya ini bisa dari ini aja

**Review by ferdifdi — APPROVED** [review]

**Review by luthfifahlevi — APPROVED** [review]
Jgn lupa ceklist di deskripsinya hug

File: src/routes/statusPage/index.js (hunk 2 above)
Comment (HugoIr) [reply src/routes/statusPage/index.js]: done

File: src/routes/statusPage/index.js (hunk 1 above)
Comment (HugoIr) [reply src/routes/statusPage/index.js]: done

File: src/components/sideBar/index.js [hunk 3]
```diff
@@ -137,6 +137,7 @@ const SideBar = ({menuCollapse, setMenuCollapse}) => {
           <MenuItem
             icon={<FaInfoCircle />}>
```
Comment (andrew-susanto) [inline src/components/sideBar/index.js]:
```suggestion
          <MenuItem
            className={router.url === "/status-page/" && style['menu-item-active']}
            icon={<FaInfoCircle />}>
```

**Review by andrew-susanto — APPROVED** [review]

**Pull request merged**